File: scripts/remove-bg.py
import urllib.request
import io
import os
from PIL import Image
import numpy as np

# Download the logo from the blob URL
url = "https://hebbkx1anhila5yf.public.blob.vercel-storage.com/file_00000000897071f59e325b5bf6295766-2o8DjrptIRIeGj4AIPL46WpmFCq7U3.png"
print("Downloading logo...")
response = urllib.request.urlopen(url)
img_data = response.read()
print(f"Downloaded {len(img_data)} bytes")

img = Image.open(io.BytesIO(img_data)).convert("RGBA")
print(f"Image size: {img.size}")
data = np.array(img)

r, g, b, a = data[:, :, 0], data[:, :, 1], data[:, :, 2], data[:, :, 3]

# White and near-white pixels (threshold 230+)
white_mask = (r > 230) & (g > 230) & (b > 230)

# Set alpha to 0 for white pixels
data[white_mask, 3] = 0

# For light gray border pixels, create a smooth fade
light_mask = (r > 200) & (g > 200) & (b > 200) & ~white_mask
brightness = (r.astype(float) + g.astype(float) + b.astype(float)) / 3.0
alpha_values = np.clip((255 - brightness) * 4, 0, 255).astype(np.uint8)
data[light_mask, 3] = alpha_values[light_mask]

result = Image.fromarray(data)

# Save to current working directory
output_path = "logo-nobg.png"
result.save(output_path, "PNG")
print(f"Saved to: {os.path.abspath(output_path)}")
print(f"File size: {os.path.getsize(output_path)} bytes")

File: scripts/remove-bg.mjs
import sharp from "sharp";
import path from "path";

const inputPath = path.resolve("public/images/logo.png");
const outputPath = path.resolve("public/images/logo-nobg.png");

async function removeBg() {
  const image = sharp(inputPath).ensureAlpha();

  // Get raw pixel data
  const { data, info } = await image
    .raw()
    .toBuffer({ resolveWithObject: true });

  const { width, height, channels } = info;

  // Process pixels: make white/near-white pixels transparent
  for (let i = 0; i < width * height; i++) {
    const idx = i * channels;
    const r = data[idx];
    const g = data[idx + 1];
    const b = data[idx + 2];

    // If the pixel is near-white (all channels > 230), make it transparent
    if (r > 230 && g > 230 && b > 230) {
      data[idx + 3] = 0; // Set alpha to 0
    }
    // Also handle light gray edges for smoother result
    else if (r > 210 && g > 210 && b > 210) {
      // Partial transparency for anti-aliasing
      const brightness = (r + g + b) / 3;
      const alpha = Math.max(0, Math.min(255, Math.round((255 - brightness) * 5)));
      data[idx + 3] = alpha;
    }
  }

  await sharp(data, { raw: { width, height, channels } })
    .png()
    .toFile(outputPath);

  console.log(`Done! Saved to ${outputPath}`);
}

removeBg().catch(console.error);
